### ISSUE TITLE
DataTable: alignment issues when table-layout is auto

### ISSUE THREAD

**Issue by RajendraDave** [primefaces/primefaces#2788]
## 1) Environment
- PrimeFaces version:6.1
- Application server + version: JBOSS 7.1.1
- Affected browsers: Firefox,IE,Chrome

## 2) Expected behavior
Header columns and body columns should be aligned. 
Column should not be of fixed width.The width must be adjusted according to its column contents:
![expected-behaviour](https://user-images.githubusercontent.com/32572729/31327137-b5082b46-acea-11e7-8deb-e64eb250504a.png)

## 3) Actual behavior
Header columns and body columns are not aligned:
![data](https://user-images.githubusercontent.com/32572729/31327141-bb45748c-acea-11e7-9574-635ca6fd293d.png)

## 4) Steps to reproduce
Just copy sample XHTML and bean.

## 5) Sample XHTML
```xhtml
	<div style="width:50%">
                <p:dataTable var="car" value="#{dtScrollView.cars4}" scrollable="true" scrollHeight="150" tableStyle="width:auto">
                    <p:column headerText="col1">
                        <h:outputText value="#{car.id}" />
                    </p:column>
                    <p:column headerText="col2" >
                        <h:outputText value="#{car.year}" />
                    </p:column>
                    <p:column headerText="col3" >
                        <h:outputText value="#{car.brand}" />
                    </p:column>
                    <p:column headerText="col4" >
                        <h:outputText value="#{car.color}" />
                    </p:column>

                </p:dataTable>
            </div>
```

## 6) Sample bean
```java
@ManagedBean
@ViewScoped
public class ScrollView implements Serializable {

    private List<Car> cars4 = new ArrayList<Car>();

    @PostConstruct
    public void init() {
        for (int i = 0; i < 5; i++) {
            Car c = new Car();
            c.setBrand("AAAAAAAAAA" + i);
            c.setColor("AAAAAA" + i);
            c.setId("AAA" + i);
            c.setYear("AAAAAAAAAAAAAA" + i);
            cars4.add(c);
        }
    }

    public List<Car> getCars4() {
        return cars4;
    }
}
```

**Comment by cagataycivici** [comment]
table-layout:fixed is **fundamental** to make resize, scroll work so auto layout is not supported technically.

**Comment by rudi1602** [comment]
fixed by #14387
